design/overview.md

```diff
@@ -0,0 +1,145 @@
+# Design Overview
+
+Details the overall design of Formica - a management layer over fleet
+
+- _Formica_ is a genus of ants, see https://en.wikipedia.org/wiki/Formica
+- See overview.png for a more graphical view of this material
+
+## General Design Overview
+- Service running inside cluster named ```formicad```
+- Command line tool to interact with ```formicad``` named ```formicactl```
+
+### ```formicad```
+- Handles generating unit files from unit file definitions
+- Handles submitting and updating unit files
+- Supports submitting unit file definitions
+- Supports querying against the state of the unit file definition being processed
+- Supports users viewing the generated unit files
+- Fleet is source of truth - no state is held in ```formicad``` that can not be regenerated from fleet
+- Uses a task system with swappable backends
+- No user management, authentication or authorization
```

Comment:
yeah, this would be in line with a potential "trusted computing" story where we would deliver certificates already in the bootstrapping phase.